appstudio-utils/util-scripts/fetch-test-data.sh

```diff
@@ -15,12 +15,20 @@ source $(dirname $0)/lib/fetch.sh
 
 # search all task runs in a pipeline
 # write output in format $basdir/data/test/$task_name/data.json
+result_found=
 for tr in $( pr-get-tr-names $PR_NAME ); do
   data=$( tr-get-result $tr $TEST_NAME )
   if [[ ! -z "${data}" ]]; then
+      result_found=1
       task_name=$( tr-get-task-name ${tr} )
       echo "${data}" | jq > $( json-data-file test ${task_name} )
   fi
 done
 
+if [[ -z $result_found ]]; then
+  # let's put an an empty hash here to express the the idea
+  # that we looked for test results and found none
+  echo '{}' > $( json-data-file test )
```

Comment:
As mentioned, this isn't really needed. I'm about 60% sure I like the idea, so it's in the patch. Let me know if you don't like it.

Reply:
It does look weird and confusing when show-data gives you a checksum and shows no files. So that's one (albeit weak) argument in favor of doing it this way.

Reply:
I do think that it makes sense for this script to do this. Its "interface" is that after it runs, test data can be processed. The data may just be empty.